x/fees/keeper/fees.go

```diff
@@ -39,96 +37,88 @@ func (k Keeper) IterateFees(
 	defer iterator.Close()
 
 	for ; iterator.Valid(); iterator.Next() {
-		fee := k.BuildFeeInfo(
-			ctx,
-			common.BytesToAddress(iterator.Key()),
-			sdk.AccAddress(iterator.Value()),
-		)
+		var fee types.Fee
+		k.cdc.MustUnmarshal(iterator.Value(), &fee)
+
 		if handlerFn(fee) {
 			break
 		}
 	}
 }
 
 // GetFee returns Fee for a registered contract
-func (k Keeper) GetFee(ctx sdk.Context, contract common.Address) (types.Fee, bool) {
-	deployerAddress, found := k.GetDeployer(ctx, contract)
-	if !found {
-		return types.Fee{}, false
-	}
-	fee := k.BuildFeeInfo(ctx, contract, deployerAddress)
-	return fee, true
-}
-
-// BuildFeeInfo returns Fee given the contract and deployer addresses
-func (k Keeper) BuildFeeInfo(ctx sdk.Context, contract common.Address, deployerAddress sdk.AccAddress) types.Fee {
-	withdrawalAddress, hasWithdrawAddr := k.GetWithdrawal(ctx, contract)
-	fee := types.Fee{
-		ContractAddress: contract.String(),
-		DeployerAddress: deployerAddress.String(),
-	}
-	if hasWithdrawAddr {
-		fee.WithdrawAddress = withdrawalAddress.String()
-	}
-	return fee
-}
-
-// GetDeployer returns the deployer address for a registered contract
-func (k Keeper) GetDeployer(ctx sdk.Context, contract common.Address) (sdk.AccAddress, bool) {
+func (k Keeper) GetFee(
+	ctx sdk.Context,
+	contract common.Address,
+) (types.Fee, bool) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixFee)
 	bz := store.Get(contract.Bytes())
```

Comment:
might be a more meaningful variable name
```suggestion
	feeBytes := store.Get(contract.Bytes())
```